src/moi_oracle.jl

```diff
@@ -9,14 +9,29 @@ The `direction` vector has to be set in the same order of variables as the `MOI.
 """
 struct MathOptLMO{OT<:MOI.AbstractOptimizer} <: LinearMinimizationOracle
     o::OT
+    use_modify::Bool
+end
+
+function MathOptLMO(o; use_modify=true)
+    return MathOptLMO(o, use_modify)
 end
 
 function compute_extreme_point(lmo::MathOptLMO{OT}, direction::AbstractVector{T}; kwargs...) where {OT,T<:Real}
     variables = MOI.get(lmo.o, MOI.ListOfVariableIndices())
-    terms = [MOI.ScalarAffineTerm(d, v) for (d, v) in zip(direction, variables)]
-    obj = MOI.ScalarAffineFunction(terms, zero(T))
-    MOI.set(lmo.o, MOI.ObjectiveFunction{typeof(obj)}(), obj)
-    MOI.set(lmo.o, MOI.ObjectiveSense(), MOI.MIN_SENSE)
+    if lmo.use_modify == true
+        for i in eachindex(variables)
+            MOI.modify(
+                lmo.o,
+                MOI.ObjectiveFunction{MOI.ScalarAffineFunction{T}}(),
+                MOI.ScalarCoefficientChange(variables[i], direction[i]),
+            )
+        end
+    else
+        terms = [MOI.ScalarAffineTerm(d, v) for (d, v) in zip(direction, variables)]
+        obj = MOI.ScalarAffineFunction(terms, zero(T))
+        MOI.set(lmo.o, MOI.ObjectiveFunction{typeof(obj)}(), obj)
+        MOI.set(lmo.o, MOI.ObjectiveSense(), MOI.MIN_SENSE)
```

Comment:
now that I think of it, this line should be in the constructor only (not in compute_extreme_point)